### PR TITLE
SG-52178 - Add not logged in pages and banners to increase conversion

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
Add Cody CTA to the blob and repo page when user is not logged in.

[Figma Design](https://www.figma.com/file/75bhaZyepWto4l1gFUPmQe/Cody's-no-access-pages-for-logged-out-users?type=design&node-id=29-4517&t=nmsBD8ECsB9LgLnE-4)

## Ref
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/52178)

[Gitstart Ticket](https://clients.gitstart.com/sourcegraph/1/tickets/SG-52178)

## Demo
Screenshot 2023-06-02 at 11 57 37 AM

## Test Plan
1. Run application on this branch.
2. Go to the blob/repo pages as unauthenticated user
3. See that the mockup specified in attached attached Figma design is displayed.

### CODE REVIEW

File: client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                            Cody AI
+                        </div>
+                    </PageHeader.Breadcrumb>
+                </PageHeader.Heading>
+            </PageHeader>
+
+            {/* Page content */}
+            <div className={classNames('d-flex flex-row flex-wrap justify-content-center', styles.headerSection)}>
+                <div>
+                    <H1>Meet Cody, your AI assistant</H1>
```
Comment (rrhyne) [inline client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx:93]:
```suggestion
                    <H1>Sign up to get Cody, our AI assistant, free</H1>
```

File: client/web/src/repo/components/TryCodyWidget/TryCodyWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        const eventPage = type === 'blob' ? 'Blobview' : 'RepositoryPage'
+        const eventArguments = {
+            type: eventType,
+            page: eventPage,
+            description: '',
+        }
+        telemetryService.log(EventName.TRY_CODY_SIGNUP_INITIATED, eventArguments, eventArguments)
+    }
+
+    const title = type === 'blob' ? 'Sign up to get Cody, our AI assistant, free.' : 'Meet Cody, your AI assistant'
```
Comment (rrhyne) [inline client/web/src/repo/components/TryCodyWidget/TryCodyWidget.tsx:80]:
```suggestion
    const title = type === 'blob' ? 'Sign up to get Cody, our AI assistant, free' : 'Meet Cody, your AI assistant'
```

**Comment by rrhyne** [comment]
Build looks great!

A few requests on the banners here:

https://www.figma.com/file/75bhaZyepWto4l1gFUPmQe/Cody's-no-access-pages-for-logged-out-users?type=design&node-id=290-4956

Additionally:
We need to make the Cody AI main navigation item visible to logged out users.  It should point to this version of the /cody page when the user is signed out.

**Comment by gitstart-sourcegraph** [comment]
Hi @rrhyne,
With regards to your [comment here](https://www.figma.com/file/75bhaZyepWto4l1gFUPmQe?node-id=183:4938#457936751). After conducting some research on the codebase, we couldn't find any prior instances where a page's content was conditionally rendered or hidden on the airgapped instances. We would appreciate some input and ideas from your team on how we can determine if the current instance is `airgapped`.

**Comment by rrhyne** [comment]
> Hi @rrhyne, With regards to your [comment here](https://www.figma.com/file/75bhaZyepWto4l1gFUPmQe?node-id=183:4938#457936751). After conducting some research on the codebase, we couldn't find any prior instances where a page's content was conditionally rendered or hidden on the airgapped instances. We would appreciate some input and ideas from your team on how we can determine if the current instance is `airgapped`.

Apparently, this isn't possible, so no change is required. Thanks for looking into it.

**Comment by rrhyne** [comment]
QA on latest build:

Breakpoints are incorrect on 1440 monitors:
image

**Comment by rrhyne** [comment]
Other than that, the build looks great, thanks! I'll ask for a review from some folks here and then let's merge!

**Review by rrhyne — APPROVED** [review]
Looks great to me pending review from Sourcegraph engineers.

File: client/web/src/util/constants.ts [hunk 3]
```diff
@@ -12,4 +12,6 @@ export const enum EventName {
     TRY_CODY_MARKETPLACE = 'VSCodeMarketplace',
     TRY_CODY_WEB = 'TryCodyWeb',
     TRY_CODY_WEB_ONBOARDING_DISPLAYED = 'TryCodyWebOnboardingDisplayed',
+    TRY_CODY_SIGNUP_INITIATED = 'SignUpInitiated',
```
Comment (eseliger) [inline client/web/src/util/constants.ts:15]: I think we have to have cody in the event name for some tooling reasons 
```suggestion
    TRY_CODY_SIGNUP_INITIATED = 'CodySignUpInitiated',
```

File: client/web/src/util/constants.ts [hunk 4]
```diff
@@ -12,4 +12,6 @@ export const enum EventName {
...
+    SPEAK_TO_AN_ENGINEER_CTA = 'SpeakToAnEngineerCTA',
```
Comment (eseliger) [inline client/web/src/util/constants.ts:16]: should we make this also cody specific? 
```suggestion
    SPEAK_TO_AN_ENGINEER_CTA = 'SpeakToACodyEngineerCTA',
```

File: client/web/src/nav/GlobalNavbar.tsx [hunk 5]
```diff
@@ -235,13 +235,14 @@ export const GlobalNavbar: React.FunctionComponent<React.PropsWithChildren<Globa
...
                     )}
-                    {codyEnabled.chat && (
-                        <NavItem icon={CodyLogo}>
-                            <NavLink variant={navLinkVariant} to={EnterprisePageRoutes.Cody}>
-                                Cody AI
-                            </NavLink>
-                        </NavItem>
-                    )}
+                    {codyEnabled.chat ||
+                        (isSourcegraphDotCom && (
```
Comment (eseliger) [inline client/web/src/nav/GlobalNavbar.tsx:239]:
```suggestion
                    {(codyEnabled.chat ||
                        isSourcegraphDotCom) && (
```

File: client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+                            context={context}
+                            githubLabel="Github"
+                            gitlabLabel="Gitlab"
+                            withCenteredText={true}
+                            onClick={onClickCTAButton}
+                            ctaClassName={styles.authButton}
+                            iconClassName={styles.buttonIcon}
+                        />
+                    </div>
+                    <Link to="https://sourcegraph.com/sign-in?returnTo=/search">Or, continue with email</Link>
```
Comment (eseliger) [inline client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx:123]: is the returnTo to search intentional here? cc @rrhyne

**Comment by gitstart-sourcegraph** [comment]
Thanks @eseliger, suggested improvements has been added.

File: client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx (hunk 6 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx:123]: No, good catch! I think this should be:

```suggestion
                    <Link to="https://sourcegraph.com/sign-up?showEmail=true">Or, continue with email</Link>
```

**Review by rrhyne — APPROVED** [review]
Let's do this! :D

File: client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/cody/components/CodyMarketingPage/CodyMarketingPage.tsx:93]: @rrhyne do you still want this change?

**Review by eseliger — APPROVED** [review]

**Pull request merged**